src/Functions/SynonymsExtensions.cpp

```diff
@@ -0,0 +1,150 @@
+#include <Common/Exception.h>
+#include <Functions/SynonymsExtensions.h>
+
+#include <fstream>
+#include <list>
+
+#include <boost/algorithm/string.hpp>
+#include <wnb/core/wordnet.hh>
+
+namespace DB
+{
+
+namespace ErrorCodes
+{
+    extern const int LOGICAL_ERROR;
+    extern const int UNKNOWN_ELEMENT_IN_CONFIG;
+    extern const int INVALID_CONFIG_PARAMETER;
+}
+
+class PlainSynonymsExtension : public ISynonymsExtension
+{
+private:
+    using Container = std::list<std::vector<String>>;
+    using LookupTable = std::unordered_map<std::string_view, Synset *>;
+    
+    Container data;
+    LookupTable table;
+
+public:
+    PlainSynonymsExtension(const String & path)
+    {
+        std::ifstream file(path);
+        if (!file.is_open())
+            throw Exception("Cannot find synonyms extention at: " + path,
+                ErrorCodes::INVALID_CONFIG_PARAMETER);
+
+        String line;
+        while (std::getline(file, line))
+        {
+            std::vector<String> words;
+            boost::split(words, line, boost::is_any_of("\t "));
+            if (!words.empty())
+            {
+                data.emplace_back(std::move(words));
+
+                for (const auto &i : data.back()) 
```

Comment:
What is `i` here?

Reply:
`i` refers to certain word in current synonym set. Changed names for better readability.